pkg/datamover/repository_server_test.go

```diff
@@ -0,0 +1,249 @@
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	kopiacmd "github.com/kanisterio/kanister/pkg/kopia/command"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/testutil"
+)
+
+type RepositoryServerSuite struct {
+	ctx                context.Context
+	address            string
+	kopiaCacheDir      string
+	kopiaLogDir        string
+	kopiaConfigDir     string
+	tlsDir             string
+	serverHost         string
+	serverUsername     string
+	serverPassword     string
+	repositoryUser     string
+	testUsername       string
+	testUserPassword   string
+	repositoryPassword string
+	repoPathPrefix     string
+	fingerprint        string
+}
+
+const (
+	TestRepositoryEncryptionKey = "TEST_REPOSITORY_ENCRYPTION_KEY"
+)
+
+var _ = Suite(&RepositoryServerSuite{})
+
+func (rss *RepositoryServerSuite) SetUpSuite(c *C) {
+	// Check if kopia binary exists in PATH
+	if !CommandExists("kopia") {
+		c.Skip("Skipping repository server datamover unit test. Couldn't find kopia binary in the path.")
+	}
+
+	// Setting Up Repository Server Address
+	rss.address = fmt.Sprintf("%s:%s", "https://0.0.0.0", strconv.Itoa(rand.IntnRange(50000, 60000)))
+
+	// Setting Up Repository Server User Access
+	rss.serverUsername = "user@localhost"
+	rss.serverPassword = "testPassword"
+	rss.serverHost = "localhost"
+	rss.testUsername = "testuser"
+	rss.testUserPassword = rand.String(8)
+
+	// Setting Up Repository Access
+	rss.repositoryUser = "repositoryUser"
+	rss.repositoryPassword = os.Getenv(TestRepositoryEncryptionKey)
+	rss.repoPathPrefix = path.Join("kopia-int", "repository-server-datamover-test")
+
+	rss.ctx = context.Background()
+
+	// Setting Up Kopia Cache, Log and Config Dir
+	rss.kopiaCacheDir = kopiacmd.DefaultCacheDirectory
+	rss.kopiaLogDir = kopiacmd.DefaultLogDirectory
+	rss.kopiaConfigDir = kopiacmd.DefaultConfigDirectory
+
+	// Setting Up TLS Dir
+	temp := c.MkDir()
+	rss.tlsDir = filepath.Join(temp, "tls-"+rand.String(5))
+}
+
+func (rss *RepositoryServerSuite) setupKopiaRepositoryServer(c *C) {
+	// Setting Up Kopia Repository
+	contentCacheMB, metadataCacheMB := kopiacmd.GetGeneralCacheSizeSettings()
+	repoCommandArgs := kopiacmd.RepositoryCommandArgs{
+		CommandArgs: &kopiacmd.CommandArgs{
+			RepoPassword:   rss.repositoryPassword,
+			ConfigFilePath: rss.kopiaConfigDir,
+			LogDirectory:   rss.kopiaLogDir,
+		},
+		CacheDirectory:  rss.kopiaCacheDir,
+		Hostname:        rss.serverHost,
+		ContentCacheMB:  contentCacheMB,
+		MetadataCacheMB: metadataCacheMB,
+		RepoPathPrefix:  rss.repoPathPrefix,
+		Username:        rss.repositoryUser,
+		Location:        testutil.GetDefaultS3CompliantStorageLocation(),
+	}
+	// First try to connect with Kopia Repository
+	c.Log("Connecting with Kopia Repository...")
+	repoConnectCmd, err := kopiacmd.RepositoryConnectCommand(repoCommandArgs)
+	c.Assert(err, IsNil)
+	_, err = Command(c, repoConnectCmd...)
+	if err != nil {
```

Comment:
Should we have a check if the error is due to the repo doesn't exist or some other error?

Reply:
Done